api_app/db/repositories/workspaces.py

```diff
@@ -67,14 +67,18 @@ def create_workspace_item(self, workspace_input: WorkspaceInCreate, auth_info: d
 
         template = self.validate_input_against_template(workspace_input.templateName, workspace_input, ResourceType.Workspace, user_roles)
 
+        # allow for workspace template taking a single address_space or multiple address_spaces
         address_space_param = {"address_space": self.get_address_space_based_on_size(workspace_input.properties)}
+        address_spaces_param = {"address_spaces": [self.get_address_space_based_on_size(workspace_input.properties)]}
```

Comment:
I think we need to be clear what address_spaces mean.
We expect a primary one in address_space, right?
If so, maybe we can name the new field additional_address_spaces?

Reply:
It's actually all the address spaces, including the primary. Reason being that's the name of the property in terraform is a single one and needs all of them. Otherwise would have to add `address_space` to `additional_address_spaces` in the terraform, and assign that to `address_space`...

The property is actually called `address_space` - https://learn.microsoft.com/en-us/rest/api/virtualnetwork/virtual-networks/create-or-update?tabs=HTTP#addressspace . I didn't want to break existing functionality so left the existing property as is.

Thoughts?

Reply:
so this is either-or?

Reply:
Yes, so the "new" bundle I have just takes in `address_spaces`, it could also use `address_space` I guess, but I have indexed the first as `address_spaces[0]`